src/mask.c

```diff
@@ -2399,6 +2416,8 @@ int do_mask_crack(const char *extern_key)
 #endif
 		else if (options.flags & FLG_EXTERNAL_CHK)
 			ext_hybrid_fix_state();
+		else if (options.flags & FLG_HC_MASK_CHK)
+			hcmask_hybrid_fix_state();
```

Comment:
This "else" doesn't look right. You can always have an external filter AS WELL as anything else.